include/mgp_mock.py

```diff
@@ -0,0 +1,1916 @@
+
+            ```
+            graph = context.graph
+            for vertex in graph.vertices:
+            ```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return Vertices(self._graph)
+
+    def is_mutable(self) -> bool:
+        """
+        Check if the graph is mutable, i.e. if it can be modified.
+
+        Returns:
+            A `bool` value that represents whether the graph is mutable.
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+
+        Examples:
+            ```graph.is_mutable()```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return not self._graph.is_immutable()
+
+    def create_vertex(self) -> Vertex:
+        """
+        Create an empty vertex.
+
+        Returns:
+            The created `Vertex`.
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+            ImmutableObjectError: If the graph is immutable.
+
+        Examples:
+            Creating an empty vertex:
+            ```vertex = graph.create_vertex()```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._graph.is_immutable():
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        return Vertex(self._graph.create_vertex())
+
+    def delete_vertex(self, vertex: Vertex) -> None:
+        """
+        Delete the given vertex if it’s isolated, i.e. if there are no edges connected to it.
+
+        Args:
+            vertex: The `Vertex` to be deleted.
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+            ImmutableObjectError: If the graph is immutable.
+            LogicErrorError: If the vertex is not isolated.
+
+        Examples:
+            ```graph.delete_vertex(vertex)```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._graph.is_immutable():
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        if not self._graph.vertex_is_isolate(vertex.id):
+            raise LogicErrorError("Logic error.")
+
+        self._graph.delete_vertex(vertex.id)
+
+    def detach_delete_vertex(self, vertex: Vertex) -> None:
+        """
+        Delete the given vertex together with all connected edges.
+
+        Args:
+            vertex: The `Vertex` to be deleted.
+
+        Raises:
+            InvalidContextError: If the graph is not in a valid context.
+            ImmutableObjectError: If the graph is immutable.
+
+        Examples:
+            ```graph.detach_delete_vertex(vertex)```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if self._graph.is_immutable():
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        self._graph.delete_vertex(vertex.id)
```

Comment:
How is this method then different than delete_vertex when edges aren't touched?

Reply:
Both `delete_vertex` and this method call `_mgp_mock.delete_vertex`, which in turn deletes a vertex from the NetworkX graph together with all its adjacent edges if they exist.
The only difference between this method and `delete_vertex` is that `delete_vertex` only does that operation if the edge is isolated/has no adjacent edges.